File: yframework-toolkit/src/main/java/org/yframework/toolkit/json/GsonUtil.java
package org.yframework.toolkit.json;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.reflect.TypeToken;
import org.yframework.toolkit.JsonUtil;

import java.util.List;
import java.util.Map;

public enum GsonUtil implements JsonUtil
{
    INSTANCE;
    private Gson gson;

    GsonUtil()
    {
        gson = new Gson();
    }

    public Gson get()
    {
        return gson;
    }

    public String toJson(Object object)
    {
        String jsonString = null;
        if (gson != null)
        {
            jsonString = gson.toJson(object);
        }
        return jsonString;
    }

    public String toPrettyJson(Object object)
    {
        return new GsonBuilder().setPrettyPrinting().create().toJson(object);
    }

    public <T> T jsonToObject(String JsonString, Class<T> cls)
    {
        T t = null;
        if (gson != null)
        {
            t = gson.fromJson(JsonString, cls);
        }
        return t;
    }

    public <T> List<T> jsonToList(String JsonString, Class<T> cls)
    {
        List<T> list = null;
        if (gson != null)
        {
            list = gson.fromJson(JsonString, new TypeToken<List<T>>()
            {
            }.getType());
        }
        return list;
    }

    public <T> List<Map<String, T>> jsonToListMap(String JsonString)
    {
        List<Map<String, T>> list = null;
        if (gson != null)
        {
            list = gson.fromJson(JsonString, new TypeToken<List<Map<String, T>>>()
            {
            }.getType());
        }
        return list;
    }

    public <T> Map<String, T> jsonToMap(String JsonString)
    {
        Map<String, T> map = null;
        if (gson != null)
        {
            map = gson.fromJson(JsonString, new TypeToken<Map<String, T>>()
            {
            }.getType());
        }
        return map;
    }
}

File: yframework-android/framework/src/main/java/org/yframework/android/util/GsonUtil.kt
package org.yframework.android.util

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.google.gson.reflect.TypeToken


/**
 * Description: GsonUtil<br>
 * Comments Name: GsonUtil<br>
 * Date: 2019-08-19 14:34<br>
 * Author: ysj<br>
 * EditorDate: 2019-08-19 14:34<br>
 * Editor: ysj
 */
object GsonUtil {
    private val gson = Gson()

    fun toJson(`object`: Any?): String {
        return gson.toJson(`object`)
    }

    fun toPrettyJson(`object`: Any?): String {
        return GsonBuilder().setPrettyPrinting().create().toJson(`object`)
    }

    fun <T> jsonToObject(JsonString: String, cls: Class<T>): T? {
        return gson.fromJson(JsonString, cls)
    }

    fun <T> jsonToList(JsonString: String): List<T>? {
        return gson.fromJson(JsonString, object : TypeToken<List<T>>() {}.type)
    }

    fun <T> jsonToListMap(JsonString: String): List<Map<String, T>>? {
        return gson.fromJson(JsonString, object : TypeToken<List<Map<String, T>>>() {}.type)
    }

    fun <T> jsonToMap(JsonString: String): Map<String, T>? {
        return gson.fromJson(JsonString, object : TypeToken<Map<String, T>>() {}.type)
    }
}